File: src/simulation/Robot.java
package simulation;

import org.jbox2d.dynamics.Body;

import java.util.ArrayList;

/**
 * Created by colander on 1/13/17.
 * Class used to represent the "robot" in the JBox2D simulation.
 */
public class Robot {
    Body body;
    ArrayList<RobotLeg> legs;

    public Robot(Body body, ArrayList<RobotLeg> legs) {
        this.body = body;
        this.legs = legs;
    }
}

File: src/main/java/com/janboucek/crawler/simulation/Robot.kt
package com.janboucek.crawler.simulation

import org.jbox2d.dynamics.Body

/**
 * Created by colander on 1/13/17.
 * Class used to represent the "robot" in the JBox2D simulation.
 */
data class Robot(val body: Body, val legs: List<RobotLeg>)